lib/pure/complex.nim

```diff
@@ -9,78 +10,135 @@
 
 
 
+
 ## This module implements complex numbers.
-{.push checks:off, line_dir:off, stack_trace:off, debugger:off.}
-# the user does not want to trace a part
-# of the standard library!
+## Complex numbers are currently implemented as generic on a 64-bit or 32-bit float.
+##
+## Converters are defined to implicitly convert ordinary numbers and complex
+## types into complex types. This works normally for 128-bit complex numbers but
+## due to type conversion being non-transitive, dividing integer literals by
+## 64-bit complex numbers and vice versa can cause compilation errors
 
+{.push checks: off, line_dir: off, stack_trace: off, debugger: off.}
+# the user does not want to trace a part of the standard library!
 
-import
-  math
 
-const
-  EPS = 1.0e-7 ## Epsilon used for float comparisons.
+import math
 
 type
-  Complex* = tuple[re, im: float]
-    ## a complex number, consisting of a real and an imaginary part
+  Complex*[T] = tuple[re, im: T]
+    ## A complex number, consisting of a real and an imaginary part.
+  Complex128* = Complex[float64]
+    ## Alias for a pair of 64-bit floats.
+  Complex64* = Complex[float32]
+    ## Alias for a pair of 64-bit floats.
+
 
 const
-  im*: Complex = (re: 0.0, im: 1.0)
+  im*: Complex128 = (re: 0.0, im: 1.0)
     ## The imaginary unit. √-1.
 
-proc toComplex*(x: SomeInteger): Complex =
-  ## Convert some integer ``x`` to a complex number.
-  result.re = x
+
+converter toComplex128*[T: SomeInteger | float64](x: T): Complex128 =
```

Comment:
IMO these converters should be removed or be more restricted, for the same reason that there is no `int` to `float` converters. Besides, `Complex[int]` is a valid type, but the converters will convert integers to `Complex[float32]` or `Complex[float64]`.

Reply:
> Make it an explicit proc

Something like it was done in [7947 here](https://github.com/nim-lang/Nim/blob/c90510ae8be92d89ddc571adbb993f190a956f58/lib/pure/complex.nim#L44)?

Reply:
> IMO these converters should be removed or be more restricted

> The restriction `SomeInteger | float64` is weird.

This is from the original PR. Nobody complained there so I kept it. But if you ask me, `Complex` should use just floats.